core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1704,13 +1704,13 @@ private void handleAuthentication() throws PeerIsSlowToWriteException, PeerIsSlo
         try {
             r = authenticator.handleIO();
             if (r == Authenticator.OK) {
-                CharSequence principal = authenticator.getPrincipal();
-                SecurityContext securityContext = securityContextFactory.getInstance(principal, SecurityContextFactory.PGWIRE);
+                SecurityContext securityContext = securityContextFactory.getInstance(authenticator.getPrincipal(), SecurityContextFactory.PGWIRE);
                 try {
                     securityContext.authorizePGWire();
```

Comment:
If this line throws there can be leak of `securityContext`

Reply:
pushed a fix